uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/mqtt311/client/sdkmqtt/Mqtt311ConnectionImpl.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.testing.mqtt311.client.sdkmqtt;
+
+import com.aws.greengrass.testing.mqtt5.client.MqttConnection;
+import com.aws.greengrass.testing.mqtt5.client.MqttLib;
+import com.aws.greengrass.testing.mqtt5.client.exceptions.MqttException;
+import com.aws.greengrass.testing.util.SslUtil;
+import lombok.NonNull;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.paho.client.mqttv3.IMqttClient;
+import org.eclipse.paho.client.mqttv3.IMqttToken;
+import org.eclipse.paho.client.mqttv3.MqttClient;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.net.ssl.SSLSocketFactory;
+
+/**
+ * Implementation of MQTT 3.1.1 connection based on AWS IoT SDK.
+ */
+public class Mqtt311ConnectionImpl implements MqttConnection {
+    private static final Logger logger = LogManager.getLogger(Mqtt311ConnectionImpl.class);
+    private static final String EXCEPTION_WHEN_CONNECTING = "Exception occurred during connect";
+    private static final String EXCEPTION_WHEN_CONFIGURE_SSL_CA = "Exception occurred during SSL configuration";
+
+    private final AtomicBoolean isClosing = new AtomicBoolean();
+    private final AtomicBoolean isConnected = new AtomicBoolean();
```

Comment:
unused?

Reply:
the variable will be used in the next task